### PR TITLE
com.diffplug.spotless 5.0

### PR DESCRIPTION
**Pull request by nedtwigg**
This PR closes out #600.  It's much simpler than the long list of commits looks.  All I did was:

- delete everything which was deprecated (had already been logging deprecation warnings)
- renamed `SpotlessExtensionBase` -> `SpotlessExtension`, and `SpotlessExtension(Modern -> Impl)`
- renamed `SpotlessTaskBase` -> `SpotlessTask`, and `SpotlessTask(Modern -> Impl)`
- changed plugin id to `com.diffplug.spotless` (lets us set new tags)
- old plugin id `com.diffplug.gradle.spotless` now gives this error message:

```
We have moved from 'com.diffplug.gradle.spotless'
                to 'com.diffplug.spotless'
To migrate:
- Test your build with: id 'com.diffplug.gradle.spotless' version '4.5.1'
- Fix any deprecation warnings (shouldn't be many / any)
- Now you can use:      id 'com.diffplug.spotless' version '5.0.0'

That's all you really need to know, but as always, there are more details in the changelog:
https://github.com/diffplug/spotless/blob/main/plugin-gradle/CHANGES.md

While you're at it, you might want to search for "target '**/".  We used
to  recommend that in our README, but it's a lot slower than something
more specific like "target 'src/**".  Also, if you haven't tried them yet,
take a look at our IDE integration and 'ratchetFrom'.  We've found them
to be useful, hope you do too.

If you like the idea behind 'ratchetFrom', you should checkout spotless-changelog
https://github.com/diffplug/spotless-changelog
```

I'm open to feedback on anything, but I think the main decision points are:

- [ ] the id migration described above, and what we want to tell users
- [ ] the decision to keep `SpotlessExtension` and `SpotlessTask` in a separate API vs impl piece.

We could smush them back together, but I thought that the "spotlessModern" refactor where we had both implementations side-by-side was useful.  Since we did the work to split them up, we might as well keep them that way, but I'm open to other thoughts.

### CODE REVIEW

**Comment by nedtwigg** [comment]
Edit: 7/7:20: added recommendation to the redirect message above about the bad performance of "target '**/"

**Comment by nedtwigg** [comment]
This is a big PR, and people are busy, no worries :). I plan to merge and release this tonight.  If you would like me to delay, feel free to ask for more time.  The code change is very simple, just deleting deprecated things.  The main thing that would benefit from a second pair of eyes is the migration message above, but having sat on it for a week, I feel pretty good about it.

**Comment by thc202** [comment]
Shouldn't the `**5.x preview:** ` note be removed now?

File: plugin-gradle/src/test/java/com/diffplug/gradle/spotless/GradleIntegrationHarness.java [hunk 1]
```diff
@@ -31,47 +31,22 @@
...
 import com.diffplug.common.tree.TreeStream;
-import com.diffplug.spotless.JreVersion;
-import com.diffplug.spotless.LineEnding;
+import com.diffplug.spotless.FileSignature;
 import com.diffplug.spotless.ResourceHarness;
 
 public class GradleIntegrationHarness extends ResourceHarness {
 	public enum GradleVersionSupport {
-		LEGACY("2.14"), KOTLIN("4.0"), CONFIG_AVOIDANCE("4.9"), MODERN(SpotlessPluginModern.MINIMUM_GRADLE), SETTINGS_PLUGINS("6.0");
+		MINIMUM("5.4"), SETTINGS_PLUGINS("6.0");
```
Comment (thc202) [inline plugin-gradle/src/test/java/com/diffplug/gradle/spotless/GradleIntegrationHarness.java:41]: `"5.4"` → `SpotlessPlugin.MINIMUM_GRADLE`?

Comment (nedtwigg) [reply plugin-gradle/src/test/java/com/diffplug/gradle/spotless/GradleIntegrationHarness.java]: Good call.

**Comment by nedtwigg** [comment]
> Shouldn't the **5.x preview:** note be removed now? And the (TODO)?

Yes, good catch!  Thanks!

**Comment by nedtwigg** [comment]
Released, thanks @thc202!

**Comment by bigdaz** [comment]
Great job on all of the Spotless improvements for Spotless 5.x. It's great that we can now leverage the newer features of Gradle. Thanks @nedtwigg !!!

**Comment by bigdaz** [comment]
Oh, and thanks for taking on the work to make the task inputs relocatable. I've done some rudimentary testing and it looks good so far. I'll report back any issues that I see in the wild.

**Comment by nedtwigg** [comment]
Thanks for the help getting us here!  The new task layout you implemented is much better.

**Pull request merged**